zsh/functions/whats-in-port

```diff
@@ -0,0 +1,23 @@
+# List process running on provided port
+#
+# whats_in_port 3000
+#
+# output:
+# COMMAND   PID   USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
+# ruby    25583   root   11u  IPv4 0xee20607697a79bf7      0t0  TCP *:irdmi (LISTEN)
+
+whats_in_port() {
+  if [ "$1" ]; then
+    lsof -n -i4TCP:"$1"
+    return
+  fi
+  echo "Port number needed to determine accessing process"
+}
+
+# Kills the process running on the provided port
+#
+# clear_port 3000
+
+clear_port() {
+  kill "$(whats_in_port "$1")"
```

Comment:
How does this work?

```
~% lsof -n -i4TCP:3000
COMMAND    PID USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
ruby    315558 mike   12u  IPv4 8405941      0t0  TCP 127.0.0.1:3000 (LISTEN)
```

That doesn't look like what I pass to `kill(1)`.

Reply:
I think this is missing the `-t` parameter:

```
~% lsof -n -i4TCP:3000 -t
31558
```

Reply:
The [-t flag](https://linux.die.net/man/8/lsof) might be helpful here:

> -t
>    This option specifies that lsof should produce terse output with process identifiers only and no header - e.g., so that the output may be piped to **kill**(1). This option selects the -w option.

Reply:
Yup great catch everyone, I had this function on another file and decided against including it 😅